drivers/accel/adxl313/iio_adxl313.c

```diff
@@ -0,0 +1,1200 @@
+			vals[0] = adxl313_iio_scale_table[0];
+			iio_format_value(buffer, sizeof(buffer), IIO_VAL_FRACTIONAL, 2, vals);
+			strcat(buf, buffer);
+			/* FULL_RES and 10-bit resolution if range larger than minimum value */
+			if (adxl313->range != ADXL313_0_5G_RANGE) {
+				strcat(buf, " ");
+				vals[0] = adxl313_iio_scale_table[adxl313->range];
+				iio_format_value(buffer, sizeof(buffer), IIO_VAL_FRACTIONAL, 2, vals);
+				strcat(buf, buffer);
+			}
+			break;
+		case ID_ADXL314:
+			vals[0] = adxl314_iio_scale;
+			iio_format_value(buffer, sizeof(buffer), IIO_VAL_FRACTIONAL, 2, vals);
+			strcat(buf, buffer);
+			break;
+		default:
+			return -EINVAL;
+		}
+		return strlen(buf);
+	} else {
+		return -EINVAL;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Reads the number of given samples for the selected channels
+ *
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param samples - Number of samples to be returned
+ *
+ * @return ret    - Result of the reading procedure.
+ * 					In case of success, the size of the read data is returned.
+*******************************************************************************/
+static int adxl313_iio_read_samples(void* dev, int* buff, uint32_t samples)
+{
+	int16_t data_x;
+	int16_t data_y;
+	int16_t data_z;
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	for(uint32_t i = 0; i < samples*iio_adxl313->no_of_active_channels;) {
+		adxl313_get_raw_xyz(adxl313, &data_x, &data_y, &data_z);
+
+		if (iio_adxl313->active_channels & NO_OS_BIT(0)) {
+			buff[i] = no_os_sign_extend32(data_x, 12);
+			i++;
+		}
+		if (iio_adxl313->active_channels & NO_OS_BIT(1)) {
+			buff[i] = no_os_sign_extend32(data_y, 12);
+			i++;
+		}
+		if (iio_adxl313->active_channels & NO_OS_BIT(2)) {
+			buff[i] = no_os_sign_extend32(data_z, 12);
+			i++;
+		}
+	}
+
+	return samples;
+}
+
+/***************************************************************************//**
+ * @brief Updates the number of active channels and the total number of
+ * 		  active channels
+ *
+ * @param dev  - The iio device structure.
+ * @param mask - Mask of the active channels
+ *
+ * @return ret - Result of the updating procedure.
+*******************************************************************************/
+static int adxl313_iio_update_channels(void* dev, uint32_t mask)
+{
+	struct adxl313_iio_dev *iio_adxl313;
+	uint8_t counter = 0;
+
+	if (!dev)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev;
+
+	iio_adxl313->active_channels = mask;
+
+	if (mask & NO_OS_BIT(0))
+		counter ++;
+	if (mask & NO_OS_BIT(1))
+		counter ++;
+	if (mask & NO_OS_BIT(2))
+		counter ++;
+	if (mask & NO_OS_BIT(3))
+		counter ++;
```

Comment:
there are only 3 channels used, the mask will never have b3 set. you may remove these lines.